orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/HttpClientUtils.java

```diff
@@ -160,9 +162,32 @@ public String httpGetAsString(String url) throws IOException {
     }
   }
 
+  private int getProxyPort() {
```

Comment:
I like that you decided to improve the existing code. It's much safer than it was before, thanks!